gen/slice_of.go

```diff
@@ -7,11 +7,13 @@ import (
 )
 
 // SliceOf generates an arbitrary slice of generated elements
+// genParams.MaxSize sets an (exclusive) upper limit on the size of the slice
+// genParams.MinSize sets an (inclusive) lower limit on the size of the slice
 func SliceOf(elementGen gopter.Gen) gopter.Gen {
 	return func(genParams *gopter.GenParameters) *gopter.GenResult {
 		len := 0
-		if genParams.Size > 0 {
-			len = genParams.Rng.Intn(genParams.Size)
+		if genParams.MaxSize > 0 || genParams.MinSize > 0 {
+			len = genParams.Rng.Intn(genParams.MaxSize-genParams.MinSize) + genParams.MinSize
```

Comment:
Yeah, I've been thinking about this as well, `Intn` panics on negative numbers. What kind of behaviour do you want if the check fails? This isn't really a context where we return errors can be returned. I can panic with a helpful message?